integration/helpers.go

```diff
@@ -831,6 +831,77 @@ func (i *TeleInstance) StartApp(conf *service.Config) (*service.TeleportProcess,
 	return process, nil
 }
 
+func (i *TeleInstance) StartApps(configs []*service.Config) ([]*service.TeleportProcess, error) {
+	type result struct {
+		process *service.TeleportProcess
+		tmpDir  string
+		err     error
+	}
+
+	results := make(chan result, len(configs))
+	for _, conf := range configs {
+		go func(cfg *service.Config) {
+			dataDir, err := os.MkdirTemp("", "cluster-"+i.Secrets.SiteName)
```

Comment:
Can this logic reuse `StartApp` above? Seems like most of logic here is the same.

Reply:
I initially tried to launch `StartApp` in a goroutine for each app, but it turns out `TeleInstance` is racy. StartApp, StartNode, StartDatabase, etc all update `i.tempDirs` and `i.Nodes` without using a mutex. I opted to add `StartApps` and handle updating the `TeleInstance` fields in a way that wasn't racy instead of adding a mutex to `TeleInstance` and finding all the possible races.

The downside is as you mentioned `StartApp` and `StartApps` are nearly identical. If you think it would be better to remove `StartApps` and go down the mutex route I'm happy to make the change.